Core/Core/MobileCore.cs

```diff
@@ -290,7 +290,8 @@ public ServiceConfiguration[] GetServiceConfigurationByType(String type)
         /// <returns>a single ServiceConfiguration</returns>
         public ServiceConfiguration GetFirstServiceConfigurationByType(String type)
         {
-            return servicesConfig.ContainsKey(type) ? servicesConfig[type] : null;
+            var lowerCaseType = type.ToLower();
```

Comment:
Instead of doing this, it would be better to make the dictionary case insensitive by changing this to:
private Dictionary<String, ServiceConfiguration> values = new Dictionary<string, ServiceConfiguration>(StringComparer.OrdinalIgnoreCase);

Reply:
Do you mean changing [this](https://github.com/aerogear/aerogear-xamarin-sdk/blob/master/Core/Core/MobileCore.cs#L75) to use the `StringComparer.OrdinalIgnoreCase`?

Reply:
yup

Reply:
I don't think we want to save the type directly as lowercase, I think we just want to do the lookup as lowercase in case the type changes in the future

Reply:
Changing that dictionary to `new Dictionary<string, ServiceConfiguration>(StringComparer.OrdinalIgnoreCase);`just makes it case insensitive, it doesn't save the type as lowercase.